pkg/loader/load.go

```diff
@@ -430,16 +431,19 @@ func (s *loaderImpl) execDMLs(dmls []*DML) error {
 
 // Run will quit when meet any error, or all the txn are drained
 func (s *loaderImpl) Run() error {
+	txnManager := newTxnManager(1024)
 	defer func() {
 		log.Info("Run()... in Loader quit")
 		close(s.successTxn)
+		txnManager.Close()
 	}()
 
 	batch := fNewBatchManager(s)
+	input := txnManager.put(s.input)
 
 	for {
 		select {
-		case txn, ok := <-s.input:
```

Comment:
will it slow if only change s.input to be no buffer without using `txnManager` here ?

Reply:
![image](https://user-images.githubusercontent.com/21104942/65104032-a71d0f00-da02-11e9-82e8-5a44912eee6e.png)
Yes. From 10:27 on I set `loader.input` a 8 size buffer and don't use `txnManager` here. The Update Event drop from 7k+ to 3k+.